datashape/coretypes.py

```diff
@@ -912,7 +914,37 @@ def __repr__(self):
         return 'dshape(%s)' % strs
 
 
-class Record(CollectionPrinter, Mono):
+class RecordMeta(type):
+    @staticmethod
+    def _unpack_slice(s, idx):
+        if not isinstance(s, slice):
+            raise TypeError(
+                'invalid field specification at position %d.\n'
+                'fields must be formatted like: {name}::{type}' % idx,
+            )
+
+        name, type_ = packed = s.start, s.step
+        if name is None:
+            raise TypeError('missing field name at position %d' % idx)
+        if type_ is None and s.stop is None:
+            raise TypeError(
+                "missing type for field '%s' at position %d" % (name, idx))
+        if s.stop is not None:
+            raise TypeError(
+                "unexpected slice stop for field '%s' at position %d.\n"
+                "hint: you might be missing a second ':'" % (name, idx),
+            )
+
+        return packed
+
+    def __getitem__(self, types):
+        if not isinstance(types, tuple):
+            types = types,
+
+        return self(map(self._unpack_slice, types, range(len(types))))
+
+
+class Record(RecordMeta('_', (CollectionPrinter, Mono), {})):
```

Comment:
any reason not to set `__metaclass__ = RecordMeta` in the class body of `Record`?